src/flint/types/fmpq_mpoly.pyx

```diff
@@ -0,0 +1,1051 @@
+        for arg in args_fmpq:
+            if arg is NotImplemented:
+                raise TypeError(f"Cannot coerce argument ('{arg}') to fmpq")
+
+        # Partial application with args in Z. We evaluate the polynomial one variable at a time
+        res = create_fmpq_mpoly(self.ctx)
+
+        fmpq_mpoly_set(res.val, self.val, self.ctx.val)
+        for (i, _), arg in zip(partial_args, args_fmpq):
+            if fmpq_mpoly_evaluate_one_fmpq(res.val, res.val, i, (<fmpq>arg).val, self.ctx.val) == 0:
+                raise ValueError("Unreasonably large polynomial")  # pragma: no cover
+        return res
+
+    def compose(self, *args) -> fmpq_mpoly:
+        """
+        Compose this polynomial with other fmpq_mpolys. Takes positional arguments.
+        """
+        cdef:
+            fmpq_mpoly res
+            fmpq_mpoly_ctx res_ctx
+            fmpq_mpoly_vec C
+            slong i, nvars = self.ctx.nvars(), nargs = len(args)
+
+        if nargs < nvars:
+            raise ValueError("Not enough arguments provided")
+        elif nargs > nvars:
+            raise ValueError("More arguments provided than variables")
+        elif not all(typecheck(arg, fmpq_mpoly) for arg in args):
+            raise TypeError("All arguments must be fmpq_mpolys")
+
+        res_ctx = (<fmpq_mpoly> args[0]).ctx
+        if not all((<fmpq_mpoly> args[i]).ctx is res_ctx for i in range(1, len(args))):
+            raise IncompatibleContextError("All arguments must share the same context")
+
+        C = fmpq_mpoly_vec(args, self.ctx, double_indirect=True)
+        res = create_fmpq_mpoly(self.ctx)
+        if fmpq_mpoly_compose_fmpq_mpoly(res.val, self.val, C.double_indirect, self.ctx.val, res_ctx.val) == 0:
+            raise ValueError("Unreasonably large polynomial")  # pragma: no cover
+        return res
+
+    def context(self):
+        """
+        Return the context object for this polynomials.
+
+            >>> from flint import Ordering
+            >>> ctx = fmpq_mpoly_ctx.get_context(2, Ordering.lex, 'x')
+            >>> p = ctx.from_dict({(0, 1): 2})
+            >>> ctx is p.context()
+            True
+        """
+        return self.ctx
+
+    def is_one(self):
+        return fmpq_mpoly_is_one(self.val, self.ctx.val)
+
+    def coefficient(self, slong i):
+        """
+        Return the coefficient at index `i`.
+
+            >>> from flint import Ordering
+            >>> ctx = fmpq_mpoly_ctx.get_context(2, Ordering.lex, 'x')
+            >>> p = ctx.from_dict({(0, 1): 2, (1, 1): 3})
+            >>> p.coefficient(1)
+            2
+        """
+        cdef fmpq v
+        if not 0 <= i < fmpq_mpoly_length(self.val, self.ctx.val):
+            raise IndexError("term index out of range")
+        else:
+            v = fmpq.__new__(fmpq)
+            fmpq_mpoly_get_term_coeff_fmpq(v.val, self.val, i, self.ctx.val)
+            return v
+
+    def exponent_tuple(self, slong i):
```

Comment:
Given that there is `monoms` perhaps the name here should be similar:
```python
In [45]: [p.exponent_tuple(i) for i in range(2)] == p.monoms()
Out[45]: True
```

Reply:
I've renamed `exponent_tuple` to `monomial`, inline with the `coefficient` function.

It may be my inexperience with SymPy and python-flint but from my perspective as a user I would expect the mathematically named functions to not return python representations of those things. For example
- `monomial` returns a tuple of exponent values, not a monomial in the form of a `fmp[zq]_mpoly`.
- `monoms` returns a list of tuples of exponent values, not a sequence of monomials in the form of `fmp[zq]_mpoly`s.
- `terms` returns a zip of coefficients and tuples of exponent values, not a sequence of terms in the form of `fmp[zq]_mpoly`s.

Given that the flint interface provides easy methods for this I think that they would be reasonable functions to have do so